climakitae/explore/warming.py

```diff
@@ -31,179 +40,223 @@
 xr.set_options(keep_attrs=True)  # Keep attributes when mutating xr objects
 
 
-def _get_postage_data(self):
+class WarmingLevels:
+    """A container for all of the warming levels-related functionality:
+    - A pared-down Select panel, under "choose_data"
+    - a "calculate" step where most of the waiting occurs
+    - an optional "visualize" panel, as an instance of WarmingLevelVisualize
+    - postage stamps from visualize "main" tab are accessible via "gwl_snapshots"
+    - data sliced around gwl window retrieved from "sliced_data"
     """
-    This function pulls data from the catalog and reads it into memory
 
-    Parameters
-    ----------
-    self: WarmingLevelParameters
-        object holding user's selections
+    catalog_data = xr.DataArray()
```

Comment:
Is there a reason for including these variables outside of the `init` method?

Reply:
Not necessarily... that's where params are defined but these ones don't have to be.

Reply:
(feel free to change it yourself)